src/py/flwr/server/state/state_test.py

```diff
@@ -0,0 +1,421 @@
+
+        actual_task_ins = task_ins_list[0]
+
+        assert actual_task_ins.task_id == task_ins.task_id  # pylint: disable=no-member
+        assert actual_task_ins.task is not None
+
+        actual_task = actual_task_ins.task
+
+        assert actual_task.created_at != ""
+        assert actual_task.delivered_at != ""
+        assert actual_task.ttl != ""
+
+        assert datetime.fromisoformat(actual_task.created_at) > datetime(
+            2020, 1, 1, tzinfo=timezone.utc
+        )
+        assert datetime.fromisoformat(actual_task.delivered_at) > datetime(
+            2020, 1, 1, tzinfo=timezone.utc
+        )
+        assert datetime.fromisoformat(actual_task.ttl) > datetime(
+            2020, 1, 1, tzinfo=timezone.utc
+        )
+
+    def test_store_delete_tasks(self) -> None:
+        """Test store_task_ins."""
+
+        # Prepare
+        consumer_node_id = 1
+        state = self.state_factory()
+        task_ins_0 = create_task_ins(consumer_node_id=consumer_node_id, anonymous=False)
+        task_ins_1 = create_task_ins(consumer_node_id=consumer_node_id, anonymous=False)
+        task_ins_2 = create_task_ins(consumer_node_id=consumer_node_id, anonymous=False)
+
+        # Insert three TaskIns
+        task_id_0 = state.store_task_ins(task_ins=task_ins_0)
+        task_id_1 = state.store_task_ins(task_ins=task_ins_1)
+        task_id_2 = state.store_task_ins(task_ins=task_ins_2)
+
+        assert task_id_0
+        assert task_id_1
+        assert task_id_2
+
+        # Get TaskIns to mark them delivered
+        _ = state.get_task_ins(node_id=consumer_node_id, limit=None)
+
+        # Insert one TaskRes and retrive it to mark it as delivered
+        task_res_0 = create_task_res(
+            producer_node_id=100, anonymous=False, ancestry=[str(task_id_0)]
+        )
+
+        _ = state.store_task_res(task_res=task_res_0)
+        _ = state.get_task_res(task_ids=set([task_id_0]), limit=None)
+
+        # Insert one TaskRes, but don't retrive it
+        task_res_1: TaskRes = create_task_res(
+            producer_node_id=100, anonymous=False, ancestry=[str(task_id_1)]
+        )
+        _ = state.store_task_res(task_res=task_res_1)
+
+        # Situation now:
+        # - State has three TaskIns, all of them delivered
+        # - State has two TaskRes, one of the delivered, the other not
+
+        assert state.num_task_ins() == 3
+        assert state.num_task_res() == 2
+
+        # Execute
+        state.delete_tasks(task_ids=set([task_id_0, task_id_1, task_id_2]))
+
+        # Assert
+        assert state.num_task_ins() == 2
+        assert state.num_task_res() == 1
+
+    # Init tests
+    def test_init_state(self) -> None:
+        """Test that state is innitialized correctly."""
+
+        # Execute
+        state = self.state_factory()
+
+        # Assert
+        assert isinstance(state, State)
+
+    # TaskIns tests
+    def test_task_ins_store_anonymous_and_retrieve_anonymous(self) -> None:
```

Comment:
Isn't this almost the same as the test `test_get_task_ins_anonymous` on line 54? We can probably merge them

Reply:
True. merged them.